### ISSUE TITLE
Username is preceeded with "/"

### ISSUE THREAD

**Issue by jason-a69** [ozomer/node-red-contrib-mongodb2#7]
Hi there,

When I type in my authentication details, my username is proceeded with "/". In this example, my username to the database is pirobot

Version details
├─┬ mongodb@2.1.21
│ ├── es6-promise@3.0.2
│ ├─┬ mongodb-core@1.3.21
│ │ ├── bson@0.4.23
│ │ └─┬ require_optional@1.0.0
│ │   ├── resolve-from@2.0.0
│ │   └── semver@5.1.0
│ └─┬ readable-stream@1.0.31
│   ├── core-util-is@1.0.2
│   ├── inherits@2.0.1
│   ├── isarray@0.0.1
│   └── string_decoder@0.10.31

Remote server mongod.log file (doing the write)
2016-06-02T11:37:33.125+0100 I NETWORK  [initandlisten] connection accepted from 192.168.0.5:40017 #5 (1 connection now open)
2016-06-02T11:37:37.539+0100 I ACCESS   [conn5] SCRAM-SHA-1 authentication failed for /pirobot on testiot from client 192.168.0.5 ; UserNotFound: Could not find user /pirobot@testiot
2016-06-02T11:37:39.375+0100 I NETWORK  [conn5] end connection 192.168.0.5:40017 (0 connections now open)

I managed to get round the issue by renaming my user to "/pirobot" but it just seems a bit odd.

Any advice is appreciated, I am happy to provide more feedback.

Many thanks

Jason

**Comment by ozomer** [comment]
Maybe you are using a URI with three slashes: `mongodb:///host/...` instead of `mongodb://host/...`.
Are you using the username and password fields, or just the URI field?

**Comment by jason-a69** [comment]
Hi Oren,

This is my configuration

Can I pass the username / password in the URI? Like this?
mongodb://pirobot:<password>@192.168.0.230:27017/testiot

Thanks for your time

Jason

**Comment by ozomer** [comment]
Yes, you can.

**Comment by jason-a69** [comment]
I just tried that, same problem

2016-06-09T10:25:11.312+0100 I NETWORK  [initandlisten] connection 
accepted from 192.168.0.4:60412 #3329 (1 connection now open)
2016-06-09T10:25:20.698+0100 I ACCESS   [conn3329] Successfully 
authenticated as principal /pirobot on testiot

**Comment by ozomer** [comment]
This is very strange. You are saying it tries to login as "/pirobot" instead of "pirobot"?
Please make sure that the username and password fields are empty, and only the URI field is filled.
The code simply calls: `mongodb.MongoClient.connect(this.uri, this.options || {})`.
I suggest you try calling `mongodb.MongoClient.connect` from node shell to see what happens.

**Comment by jason-a69** [comment]
Yes it tried to connect as "/pirobot" instead of "pirobot"

So to debug this, I need to run a shell, with the command node, and try 
that connect statement?

**Comment by ozomer** [comment]
Yes, try to connect normally with a [mongodb](https://www.npmjs.com/package/mongodb) client and see what happens.

**Comment by jason-a69** [comment]
That works fine, very strange
